### ISSUE TITLE
AllureLifecycle method getCurrentTestCase() returns UUID of step

### ISSUE THREAD

**Issue by Sletov** [allure-framework/allure-java#373]
#### I'm submitting

- [x] bug report

#### Current behavior:
_Allure.getLifecycle().getCurrentTestCase()_ returns UUID of current running  **test step**

#### Steps to reproduce:
Calling _Allure.getLifecycle().getCurrentTestCase()_ in any Cucumber step (or hook) returns UUID of the step. 

#### Expected behavior:
_Allure.getLifecycle().getCurrentTestCase()_ returns UUID of current running **test case**

#### Motivation / use case for changing the behavior:
It's impossible to update current test case (write steps or add attachment) with method _AllureLifecycle public updateTestCase(String, Consumer<TestResult>)_. 
Method _AllureLifecycle public updateTestCase(Consumer<TestResult>)_ (for current running test case) doesn't work too.

#### Environment:

| Allure version | 2.12.1 |
| Test framework | Cucumber@4.3.1 |
| Allure integration | allure-cucumber4-jvm@2.12.1 |
| Generate report using | allure-maven@2.10.0 |

#### Other information 
I think problem is at io.qameta.allure.AllureLifecycle.java:223  (threadContext.clear() when starting fixture (hook))

**Comment by vmyasnik** [comment]
Аналогичное поведение на page-factory-2 14.0.1
```
AllureLifecycle lifecycle = Allure.getLifecycle();
System.out.println(lifecycle.getCurrentTestCase().get());
```
Вывод: 
TestCase56df8488-6468-4b16-be9f-1d320f0e84d7create new session6

То есть имя степа 

Далее 
lifecycle.updateTestCase(..)
Падает 
[ERROR] io.qameta.allure.AllureLifecycle                   - Could not update test case: test case with uuid TestCase56df8488-6468-4b16-be9f-1d320f0e84d7create new session6 not found

io.qameta.allure:allure-cucumber2-jvm:2.10.0
io.cucumber:cucumber-java:2.4.0

**Comment by sergeim-cc** [comment]
any update or workaround on this issue?

**Comment by Novikevicius** [comment]
This issue is still reproducible with io.qameta.allure:allure-testng:2.22.2. Is there an ETA when it will be fixed? Or at least please provide a workaround.

**Comment by mov2day** [comment]
Do we have any workaround to use for time being until the fix is available ?

**Comment by arshwarsi** [comment]
Due to this issue, not able to add Runtime Parameters, or Change Testcase name in report. We wanted to update testcase name to include browser for Cross Browser testing, but it is not working.

**Comment by baev** [comment]
fixed via https://github.com/allure-framework/allure-java/pull/1011

released as [2.26.0](https://github.com/allure-framework/allure-java/releases/tag/2.26.0)
